tests/phase0/helpers.py

```diff
@@ -0,0 +1,144 @@
+from copy import deepcopy
+
+from py_ecc import bls
+
+import build.phase0.spec as spec
+from build.phase0.utils.minimal_ssz import signed_root
+from build.phase0.spec import (
+    # constants
+    EMPTY_SIGNATURE,
+    # SSZ
+    AttestationData,
+    Deposit,
+    DepositInput,
+    DepositData,
+    Eth1Data,
+    # functions
+    get_block_root,
+    get_current_epoch,
+    get_domain,
+    get_empty_block,
+    get_epoch_start_slot,
+    get_genesis_beacon_state,
+    verify_merkle_branch,
+    hash,
+)
+from build.phase0.utils.merkle_minimal import (
+    calc_merkle_tree_from_leaves,
+    get_merkle_proof,
+    get_merkle_root,
+)
+
+
+privkeys_list = [i + 1 for i in range(1000)]
+pubkeys_list = [bls.privtopub(privkey) for privkey in privkeys_list]
+pubkey_to_privkey = {pubkey: privkey for privkey, pubkey in zip(privkeys_list, pubkeys_list)}
+
+
+def create_mock_genesis_validator_deposits(num_validators, deposit_data_leaves):
+    deposit_timestamp = 0
+    proof_of_possession = b'\x33' * 96
+
+    deposit_data_list = []
+    for i in range(num_validators):
+        pubkey = pubkeys_list[i]
+        privkey = pubkey_to_privkey[pubkey]
+        deposit_data = DepositData(
+            amount=spec.MAX_DEPOSIT_AMOUNT,
+            timestamp=deposit_timestamp,
+            deposit_input=DepositInput(
+                pubkey=pubkey,
+                withdrawal_credentials=privkey.to_bytes(32, byteorder='big'),
```

Comment:
It seems incorrect or just a stub. Assume that we use validator pubkeys as the withdrawal pubkeys, than it can be:

```python
        withdrawal_credentials=BLS_WITHDRAWAL_PREFIX_BYTE  + hash(pubkey)[1:],
```

or add a comment to note that it's a stub and use a made-up value e.g., `b'\x12' * 32`.

I prefer to use `BLS_WITHDRAWAL_PREFIX_BYTE` here.

Reply:
Seems like a couple of the teams (lighthouse & nimbus) put the privkey in withdrawal credentials as a convention. To do so validly, needed to go bigint to ensure the 0-byte for these lower values.

It doesn't actually buy us anything really. it might just be better to assume that clients have a rainbow table for the first 100k pubkeys to do reverse lookup